### PR TITLE
Class method alternative approach

### PR DESCRIPTION
**Pull request by Arusekk**
BTW Python is object-oriented, so OOP >>>>> nested lambdas functional stuff :wink:

Depends on https://github.com/kuznia-rdzeni/coreblocks/pull/263
See https://github.com/kuznia-rdzeni/coreblocks/pull/265

### CODE REVIEW

File: test/common.py [hunk 1]
```diff
@@ -327,68 +328,41 @@ def process(arg):
...
+        self.kwargs = kwargs
+
+    def __call__(self, func: Callable[..., Optional[RecordIntDict]]) -> Callable[[], TestGen[None]]:
         @functools.wraps(func)
-        def mock(self) -> TestGen[None]:
-            tb = tb_getter(self)
+        def mock(func_self=None, /) -> TestGen[None]:
+            f = func
+            tb_getter = self.tb_getter
+            if func_self is not None:
```
Comment (xThaid) [inline test/common.py:357]: It'd be useful to wrap other lambdas as weel. For example `condition` lambda.

Comment (xThaid) [reply test/common.py:357]: That would effectively solve issue #262

Comment (tilk) [reply test/common.py:357]: I'd leave this to another PR, this one is pretty clear and uncontroversial (other than the use of `class` instead of `def` XD)

**Comment by tilk** [comment]
If you're force-pushing anyway, why not do a proper rebase? ;)

**Comment by Arusekk** [comment]
Done. 😄

**Comment by xThaid** [comment]
I don't really know why it works, but I like the result of the change -- helps writing tests more efficiently.

**Review by xThaid — APPROVED** [review]

**Review by tilk — APPROVED** [review]

**Pull request merged**